### ISSUE TITLE
Add Levelling Strategy: 'Breeding'

### ISSUE THREAD

**Issue by lokka30** [lokka30/LevelledMobs#63]
Requested by Leasoncre on the support discord

```yaml
breeding-spawns-leveled-mobs: false #breeding your mobs will spawn leveled mobs
breeding-math: AVERAGE|RANDOM #how should the bred animal's level be determined? (Does not go over max-level-overrides)
```

**Comment by lokka30** [comment]
Should not be very difficult to implement.

I will adjust the original idea to be more fitting for the latest version of LM:

```yaml
# This setting manages the level that a baby mob receives if it was born from breeding.
# 
# Types:
#   AVERAGE: The baby will receive the mean level of the two parents.
#   GENERATE: The baby will receive a generated level, as if it were any other mob. (makes the setting to do nothing)
#   NONE: The baby will not be levelled at all.  
breeding-levelling: 'AVERAGE' 
```

**Comment by lokka30** [comment]
Suggestion by MrCrazy: also allow it so tamed mobs get their level upgraded from their non-tamed status. unsure how this will be configured, maybe a % increase from their current lvl.

**Comment by lokka30** [comment]
The oldest issue left open on the issue tracker. It'll be solved in v4.1. :)

**Comment by lokka30** [comment]
Implemented in LM 4.0 :)
